src/ice/agent.rs

```diff
@@ -775,6 +775,18 @@ impl IceAgent {
             }
         }
 
+        if message.is_successful_binding_response()
+            && !self
+                .candidate_pairs
+                .iter()
+                .any(|p| p.has_binding_attempt(message.trans_id()))
+        {
+            trace!(
+                "Message rejected, transaction ID does not belong to any of our candidate pairs"
+            );
+            return false;
+        }
+
```

Comment:
(Niggly?) style note. I like to break up iterator/combinator style expressions from the if-statement. That gives us both an opportunity to have a self-documenting variable name as well as a good place for further optional code doc, if needed.

https://github.com/algesten/str0m/blob/main/src/change/sdp.rs#L964-L978

That does give us two nested if-statements, but on balance I think that's OK in this instance.

```rs
if message.is_successful_binding_response() {
    // <optional clarifying doc if variable name isn't enough>
    let good_variable_name = !self.candidate_pairs.iter().any(…);
    
    if !good_variable_name {
        return false;
    }    
}
```

I'm not being consistent with this, it's something I increasingly try to do. The trigger for me is when an if-statement condition starts to break on to multiple lines.

Reply:
I also did this and then clippy complained that they can be collapsed into one! Do you want me to add a `#[allow(clippy::)]` for the lint or should I find a way of refactoring it such that the lint doesn't fire?

Reply:
As you can see in `lib.rs`, I'm quite happy to turn off lints. I don't like clippy getting in the way for code clarity.